src/odemis/acq/stream/__init__.py

```diff
@@ -124,10 +124,14 @@ def __len__(self):
         acc = 0
 
         for s in self.streams:
-            if isinstance(s, Stream):
+            leaf = s
```

Comment:
Just keep it simple, use the "s" variable, and: `if isinstance(leaf, (Stream, DataProjection)):`